cpp/test/matrix/gather.cu

```diff
@@ -72,10 +72,16 @@ struct GatherInputs {
   IdxT nrows;
   IdxT ncols;
   IdxT map_length;
+  IdxT col_batch_size;
   unsigned long long int seed;
 };
 
-template <bool Conditional, bool MapTransform, typename MatrixT, typename MapT, typename IdxT>
+template <bool Conditional,
+          bool MapTransform,
+          bool Inplace,
```

Comment:
Great suggestion. Not implementing this now because I am not overly concerned with the execution speed of the tests (they still take very little time compared to some other matrix tests). And while these optimizations are great, the tests are needed mainly to check the accuracy. However, I am open to suggestions if @cjnolet wants to weigh in.